README.md

```diff
@@ -640,6 +640,42 @@ note: Step 3: call itself without input consumption - left recursion
   |        ^^^^^
 ```
 
+## Plugins API
+
+Plugin is an object with the `use(config, options)` method. That method will be
+called for all plugins in the `options.plugins` array, supplied to the `generate()`
+method.
+
+Plugin accepts the following parameters:
+- `config` — object with the following properties:
+  - `parser` — `Parser` object, by default the `peg.parser` instance. That object
+    will be used to parse the grammar. Plugin can replace this object
+  - `passes` — mapping `{ [stage: string]: Pass[] }` that represents compilation
+    stages that would applied to the AST, returned by the `parser` object. That
+    mapping will contain at least the following keys:
```

Comment:
"at least" implies that plugins can add passes, which I think they can, but it will be hard for them to add them in the middle at the moment.

Reply:
Yes, you right. I use that turn of speech to show that in the future list of stages could be expanded. For example, in my implementation of template rules (pegjs/pegjs#45) I've needed a `precheck` stage. I'm not sure, that this was right choice, although. If someday I returns to that problem I review that choice.